### PR TITLE
State Trie

### PR DESCRIPTION
**Pull request by lovesh**

### CODE REVIEW

File: sovrin_client/agent/agent.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
 from plenum.common.util import randomString
 
 logger = getlogger()
-logger.setLevel(agentLoggingLevel)
+# logger.setLevel(agentLoggingLevel)
```
Comment (ashcherbakov) [inline sovrin_client/agent/agent.py]: why we commented this?

File: sovrin_client/agent/walleted.py [hunk 2]
```diff
@@ -49,7 +49,7 @@
 from plenum.common.constants import PUBKEY
 
 logger = getlogger()
-logger.setLevel(agentLoggingLevel)
+# logger.setLevel(agentLoggingLevel)
```
Comment (ashcherbakov) [inline sovrin_client/agent/walleted.py]: Why we commented this?

File: sovrin_client/client/client.py [hunk 3]
```diff
@@ -136,99 +118,29 @@ def postReplyRecvd(self, identifier, reqId, frm, result, numReplies):
...
-    def getTxnById(self, txnId: str):
-        if self.graphStore:
-            txns = list(self.graphStore.getResultForTxnIds(txnId).values())
-            return txns[0] if txns else {}
-        else:
-            # TODO: Add support for fetching reply by transaction id
-            # serTxn = self.reqRepStore.getResultForTxnId(txnId)
-            pass
-            # TODO Add merkleInfo as well
+        # if isinstance(self.reqRepStore, ClientReqRepStoreOrientDB):
```
Comment (ashcherbakov) [inline sovrin_client/client/client.py]: Please remove commented code

File: sovrin_client/client/client.py [hunk 4]
```diff
@@ -90,21 +84,8 @@ def handlePeerMessage(self, msg):
         """
         return self.peerMsgRouter.handle(msg)
 
-    def _getOrientDbStore(self):
```
Comment (ashcherbakov) [inline sovrin_client/client/client.py]: We don't use orientdb store anymore.
Why did we need it before? Should we find a (levledb) replacement?

File: sovrin_client/client/wallet/wallet.py [hunk 5]
```diff
@@ -187,35 +187,22 @@ def addLastKnownSeqs(self, identifier, seqNo):
...
-
     def pendSyncRequests(self):
-        pendingTxnsReqs = self.getPendingTxnRequests()
-        for req in pendingTxnsReqs:
-            self.pendRequest(req)
+        # pendingTxnsReqs = self.getPendingTxnRequests()
+        # for req in pendingTxnsReqs:
+        #     self.pendRequest(req)
+
+        # GET_TXNS is discontinued
```
Comment (ashcherbakov) [inline sovrin_client/client/wallet/wallet.py]: Do we plan to support GET_TXNS or it's discontinued forever? If we plan to support, then either support or create a ticket in Batching Epic. If we don't plan to support it at all, then please remove this code.

File: sovrin_client/persistence/attribute_store.py [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
-        pass
-
-    @abstractmethod
-    def getAttribute(self, name: str, dest: Optional[str] = None):
-        pass
-
-    @abstractproperty
-    def attributes(self):
-        pass
+# class AttributeStore:
```
Comment (ashcherbakov) [inline sovrin_client/persistence/attribute_store.py]: Please remove it if it's not needed

File: sovrin_client/persistence/attribute_store_file.py [hunk 7]
```diff
@@ -1,63 +1,63 @@
...
-        key = self.attrKey(name, dest)
-        value = self.store.get(key)
-        if value:
-            return self.constructAttrData(value, name, dest)
-
-    @property
-    def attributes(self):
-        return [self.constructAttrData(val, *self.attrKeyParts(key))
-                for key, val in self.store.iterator()]
+# import base64
```
Comment (ashcherbakov) [inline sovrin_client/persistence/attribute_store_file.py]: Please remove it if we don't need

File: sovrin_client/persistence/client_req_rep_store_orientdb.py [hunk 8]
```diff
@@ -1,194 +1,194 @@
...
-            format(LAST_TXN_DATA, value, f.IDENTIFIER.nm,
-                   identifier, f.IDENTIFIER.nm, identifier))
-
-    def getLastTxnForIdentifier(self, identifier):
-        result = self.store.client.command(
-            "select value from {} where {} = '{}'".
-            format(LAST_TXN_DATA, f.IDENTIFIER.nm,
-                   identifier))
-        return None if not result else result[0].oRecordData['value']
+# from typing import Any, Sequence, List
```
Comment (ashcherbakov) [inline sovrin_client/persistence/client_req_rep_store_orientdb.py]: Please delete the file if it's not needed anymore

File: sovrin_client/test/cli/test_tutorial_manual.py [hunk 9]
```diff
@@ -54,6 +54,7 @@ def testGettingStartedTutorialAgainstSandbox(newGuyCLI, be, do):
 
 
 @pytest.mark.skipif('sys.platform == "win32"', reason='SOV-384')
+@pytest.mark.skipif('sys.platform == "linux"', reason='SOV-501')
```
Comment (ashcherbakov) [inline sovrin_client/test/cli/test_tutorial_manual.py]: The test works in master. Please fix

File: sovrin_client/test/cli/test_agent_startup.py [hunk 10]
```diff
@@ -58,6 +58,9 @@ def testAgentStartedWithoutPoolStarted(emptyLooper, tdirWithPoolTxns):
     stopAgent(emptyLooper, newAgentName)
 
 
+@pytest.mark.skip('QUESTION: What is this test checking? SCHEMA and ISSUER_KEYS'
```
Comment (ashcherbakov) [inline sovrin_client/test/cli/test_agent_startup.py]: The test works in master. Please fix

File: sovrin_client/test/test_orientdb.py [hunk 11]
```diff
@@ -1,13 +1,13 @@
...
-
-def testAddSteward(nodeSet, stewardWallet, steward):
-    for node in nodeSet:
-        assert node.graphStore.hasSteward(stewardWallet.defaultId)
-
-
-def testAddTrustAnchor(nodeSet, addedTrustAnchor):
-    for node in nodeSet:
-        assert node.graphStore.hasTrustAnchor(addedTrustAnchor.defaultId)
+# from stp_core.common.log import getlogger
```
Comment (ashcherbakov) [inline sovrin_client/test/test_orientdb.py]: Please delete the file

File: sovrin_client/client/client.py (hunk 4 above)
Comment (lovesh) [inline sovrin_client/client/client.py]: No, we used to store a graph for client, long ago, then we made it optional, we can again support it but there is no point, we dont want clients to install any db

File: sovrin_client/client/wallet/wallet.py (hunk 5 above)
Comment (lovesh) [inline sovrin_client/client/wallet/wallet.py]: No

File: sovrin_client/persistence/attribute_store.py (hunk 6 above)
Comment (lovesh) [inline sovrin_client/persistence/attribute_store.py]: We might need it.

File: sovrin_client/persistence/attribute_store_file.py (hunk 7 above)
Comment (lovesh) [inline sovrin_client/persistence/attribute_store_file.py]: We might need it.

File: sovrin_client/agent/agent.py (hunk 1 above)
Comment (lovesh) [inline sovrin_client/agent/agent.py]: Uncommented.

File: sovrin_client/agent/walleted.py (hunk 2 above)
Comment (lovesh) [inline sovrin_client/agent/walleted.py]: Uncommented.

File: sovrin_client/client/client.py (hunk 3 above)
Comment (lovesh) [inline sovrin_client/client/client.py]: Remove

File: sovrin_client/test/cli/test_tutorial_manual.py (hunk 9 above)
Comment (lovesh) [inline sovrin_client/test/cli/test_tutorial_manual.py]: Done, was a merge error

**Pull request merged**